src/util/sss_ssh.c

```diff
@@ -18,13 +18,59 @@
     along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 
+#include <ctype.h>
 #include <arpa/inet.h>
 
 #include "db/sysdb.h"
 #include "util/util.h"
 #include "util/crypto/sss_crypto.h"
 #include "util/sss_ssh.h"
 
+
+#define SKIP_SPACES(p)       while (*p != '\0' &&  isspace(*p)) { p++; }
+#define SKIP_UNTIL_SPACE(p)  while (*p != '\0' && !isspace(*p)) { p++; }
+
+/* Check if the key_line we received from the backend includes the hostname or
+ * it is just the keytype and the key.
+ *
+ * Key lines have this format:
+ * marker (optional), hostnames, keytype, base64-encoded key, comment (optional)
+ *
+ * This is a very simplistic method based on looking for the provided hostname
+ * into the provided keyline, at the right position (the hostname could also
+ * be present in the comment at the end).
+ */
+static bool
+sss_ssh_key_has_host_name(const char *key_line, const char *hostname)
+{
+    const char *current = (const char *) key_line;
+    const char *end;
+    size_t len;
+
+    SKIP_SPACES(current);
+    if (*current == '@') {
+        /* If the optional marker is present, we assume the host name is present too */
+        return true;
+    }
+
+    /* We are supposed to be here at the beginning of the hostnames. Are we?
+     * Look for the next space, which is a separator. If the hostname list
+     * is present, it must happen before that space and include the expected
+     * hostname.
+     */
+    len = strlen(hostname);
+    end = current;
+    SKIP_UNTIL_SPACE(end);
+    end -= len - 1; /* Don't go up to the end if the hostname doesn't fit */
+
+    while (current < end && strncmp(current, hostname, len) != 0) {
+        current++;
+    }
```

Comment:
`memmem(current, end - current, hostname, len)`?

Reply:
and without `end -= len - 1;`

Reply:
Will do.